app/src/main/java/com/example/tasky/feature_authentication/domain/use_case/Authenticate.kt

```diff
@@ -0,0 +1,12 @@
+package com.example.tasky.feature_authentication.domain.use_case
+
+import com.example.tasky.feature_authentication.data.util.AuthResult
+import com.example.tasky.feature_authentication.domain.repository.AuthRepository
+
+class Authenticate(
+    private val repository: AuthRepository
+) {
+    suspend operator fun invoke(): AuthResult {
+        return repository.authenticate()
+    }
+}
```

Comment:
I'm personally not a big fan of such empty use cases. If you do this for the whole project, you'll end up with 15-20 of such use cases which do nothing else than calling a repository's function. Clean architecture is fine and all, but I personally think we need to adjust it to our project while keeping the benefits. I've seen so many projects where I had to click through 5+ classes and layers just to see what a function call does. That doesn't make code more readable, but much harder to understand.

Instead, you could just call the repo's function in your ViewModel which is totally fine, since the repo's abstraction sits in domain.

Again, just my personal view and experience. Other people might see that differently, so what you do with it is up to you 😄